compiler/rustc_borrowck/src/type_check/constraint_conversion.rs

```diff
@@ -137,36 +145,83 @@ impl<'a, 'tcx> ConstraintConversion<'a, 'tcx> {
-        match k1.unpack() {
-            GenericArgKind::Lifetime(r1) => {
-                let r1_vid = self.to_region_vid(r1);
-                let r2_vid = self.to_region_vid(r2);
-                self.add_outlives(r1_vid, r2_vid, constraint_category);
-            }
+        let mut outlives_predicates = vec![(predicate, constraint_category)];
+        while let Some((ty::OutlivesPredicate(k1, r2), constraint_category)) =
+            outlives_predicates.pop()
+        {
+            match k1.unpack() {
+                GenericArgKind::Lifetime(r1) => {
+                    let r1_vid = self.to_region_vid(r1);
+                    let r2_vid = self.to_region_vid(r2);
+                    self.add_outlives(r1_vid, r2_vid, constraint_category);
+                }
 
-            GenericArgKind::Type(t1) => {
-                // we don't actually use this for anything, but
-                // the `TypeOutlives` code needs an origin.
-                let origin = infer::RelateParamBound(DUMMY_SP, t1, None);
+                GenericArgKind::Type(mut t1) => {
+                    // Normalize the type we receive from a `TypeOutlives` obligation
+                    // in the new trait solver.
+                    if infcx.next_trait_solver() {
+                        let result = CustomTypeOp::new(
+                            |ocx| {
+                                match deeply_normalize(
+                                    ocx.infcx.at(
+                                        &ObligationCause::dummy_with_span(self.span),
+                                        param_env,
+                                    ),
+                                    t1,
+                                ) {
+                                    Ok(normalized_ty) => {
+                                        t1 = normalized_ty;
+                                    }
+                                    Err(e) => {
+                                        infcx.err_ctxt().report_fulfillment_errors(e);
+                                    }
+                                }
 
-                TypeOutlives::new(
-                    &mut *self,
-                    tcx,
-                    region_bound_pairs,
-                    Some(implicit_region_bound),
-                    known_type_outlives_obligations,
-                )
-                .type_must_outlive(origin, t1, r2, constraint_category);
-            }
+                                Ok(())
+                            },
+                            "normalize type outlives obligation",
+                        )
+                        .fully_perform(infcx, self.span);
+
+                        match result {
+                            Ok(TypeOpOutput { output: (), constraints, .. }) => {
+                                if let Some(constraints) = constraints {
+                                    assert!(
+                                        constraints.member_constraints.is_empty(),
```

Comment:
you don't, member constraints shouldn't be returned here i think? they are only returned by a single `CustomTypeOp` which is directly called in `fn type_check`